src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -3513,6 +3517,55 @@ impl<'a> TypeChecker<'a> {
             }
         }
 
+        let const_udf_type = if ["wasm"].contains(&udf_definition.language.to_lowercase().as_str())
+        {
+            let file_location = match udf_definition.code.strip_prefix('@') {
+                Some(location) => FileLocation::Stage(location.to_string()),
+                None => {
+                    let uri = UriLocation::from_uri(
+                        udf_definition.code.clone(),
+                        "".to_string(),
+                        BTreeMap::default(),
+                    )?;
+                    FileLocation::Uri(uri)
+                }
+            };
+
+            let (stage_info, wasm_module_path) =
+                resolve_file_location(self.ctx.as_ref(), &file_location)
+                    .await
+                    .map_err(|err| {
+                        ErrorCode::SemanticError(format!(
+                            "Failed to resolve WASM code location {:#?}: {}",
+                            &udf_definition.code, err
+                        ))
+                    })?;
+
+            let op = StageTable::get_op(&stage_info).map_err(|err| {
+                ErrorCode::SemanticError(format!("Failed to get StageTable operator: {}", err))
+            })?;
+
+            let code_blob = op.read(&wasm_module_path).await.map_err(|err| {
+                ErrorCode::SemanticError(format!(
+                    "Failed to read WASM module {}: {}",
+                    wasm_module_path.to_string(),
+                    err
+                ))
+            })?;
+
```

Comment:
Add decompression strategy.

Reply:
The necessary changes to meet the requirements have been completed, and PR is now ready for review.